pkg/networking/middleware/retry_middleware.go

```diff
@@ -0,0 +1,108 @@
+package middleware
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/cenkalti/backoff/v5"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+)
+
+const defaultRetryCount uint = 3
+const configurationKeyRetryCount = "INTERNAL_MAX_NET_RETRIES"
+const defaultRetryAfterSeconds = 5
+
+type RetryMiddleware struct {
+	next   http.RoundTripper
+	config configuration.Configuration
+}
+
+func NewRetryMiddleware(roundTriper http.RoundTripper, config configuration.Configuration) *RetryMiddleware {
```

Comment:
nitpick: I'd put configuration first (as it makes sense to have a certain order like ctx, config, rest)

Reply:
Also a small typo `roundTriper => roundTripper` 😀

Reply:
Done & Done!